### PR TITLE
[FELIX-6281] Potential class space issues for getLogger(Bundle ...)

### PR DESCRIPTION
**Pull request by rotty3000**

### CODE REVIEW

File: log/src/main/java/org/apache/felix/log/LogServiceImpl.java [hunk 1]
```diff
@@ -126,8 +125,8 @@ public Logger getLogger(String name) {
 
     @Override
     public Logger getLogger(Class<?> clazz) {
-        LogService logService = getLogService(clazz);
-        return logService.getLogger(clazz.getName());
+        Bundle bundle = FrameworkUtil.getBundle(clazz);
```
Comment (tjwatson) [inline log/src/main/java/org/apache/felix/log/LogServiceImpl.java:128]: The class here is only to get the name of the logger.  It is not intended to get a logger for another bundle.

File: log/src/main/java/org/apache/felix/log/LogServiceImpl.java [hunk 2]
```diff
@@ -137,29 +136,13 @@ public Logger getLogger(Class<?> clazz) {
 
     @Override
     public <L extends Logger> L getLogger(Class<?> clazz, Class<L> loggerType) {
-        LogService logService = getLogService(clazz);
-        return logService.getLogger(clazz.getName(), loggerType);
+        Bundle bundle = FrameworkUtil.getBundle(clazz);
```
Comment (tjwatson) [inline log/src/main/java/org/apache/felix/log/LogServiceImpl.java:139]: Same comment about the class param

File: log/src/main/java/org/apache/felix/log/LogServiceImpl.java [hunk 3]
```diff
@@ -137,29 +136,13 @@ public Logger getLogger(Class<?> clazz) {
...
-        }
-        BundleContext bundleContext = bundle.getBundleContext();
-        ServiceReference<LogService> serviceReference = bundleContext.getServiceReference(LogService.class);
-        return bundleContext.getService(serviceReference);
-    }
-
-    private LogService getLogService(Class<?> clazz) {
-        Bundle bundle = FrameworkUtil.getBundle(clazz);
-        return getLogService(bundle);
+        return m_loggerAdminImpl.getLogger(bundle, name, loggerType);
```
Comment (tjwatson) [inline log/src/main/java/org/apache/felix/log/LogServiceImpl.java:145]: I think this should work

File: log/src/main/java/org/apache/felix/log/LogServiceImpl.java (hunk 1 above)
Comment (rotty3000) [reply log/src/main/java/org/apache/felix/log/LogServiceImpl.java:128]: got it! fixing

File: log/src/main/java/org/apache/felix/log/LogServiceImpl.java (hunk 2 above)
Comment (rotty3000) [reply log/src/main/java/org/apache/felix/log/LogServiceImpl.java]: fixed

**Pull request merged**